x-pack/packages/observability/observability_utils/observability_utils_server/llm/service_rca/index.ts

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MessageRole } from '@kbn/inference-plugin/common';
+import {
+  ChatCompletionEvent,
+  ChatCompletionEventType,
+  ChatCompletionMessageEvent,
+  ToolMessage,
+  UserMessage,
+} from '@kbn/inference-plugin/common/chat_complete';
+import { catchError, from, merge, Observable, of, OperatorFunction, switchMap } from 'rxjs';
+import { ToolCallsOf, ToolChoiceType } from '@kbn/inference-plugin/common/chat_complete/tools';
+import { callTools } from '../call_tools';
+import {
+  analyzeEntityHealth,
+  EntityHealthAnalysis,
+  EntityHealthAnalysisParameters,
+} from './analyze_entity_health';
+import { RCA_SYSTEM_PROMPT_BASE } from './system_prompt_base';
+import { findRelatedEntitiesViaKeywordSearches } from './find_related_entities_via_keyword_searches';
+import { writeKeywordSearch } from './write_keyword_search';
+import { writeRcaReport } from './write_rca_report';
+import { generateSignificantEventsTimeline, SignificantEventsTimeline } from './generate_timeline';
+
+const tools = {
```

Comment:
Could you please explain the concept of tools? 

How do tools differ from function calling?

What does the schema for tools represent?

Could you provide an explanation of how tools are called, what data is available to a tool, and how to register new tools? Or point us in the direction of any documentation.

Reply:
tools are function calling! we have docs here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability_ai_assistant/README.md